src/msg.jl

```diff
@@ -151,7 +160,16 @@ end
 function trysetproperty!(s::Message, p::Symbol, v)
   hasfield(typeof(s), p) || return s
   ftype = fieldtype(typeof(s), p)
-  setfield!(s, p, convert(ftype, v))
+  ftype === Symbol && (v = Symbol(v))
+  if v === nothing
+    ftype === Float32 && (v = NaN32)
+    ftype === Float64 && (v = NaN64)
+  end
+  try
+    setfield!(s, p, convert(ftype, v))
+  catch ex
+    @warn "Error setting field $p to $v: $ex"
```

Comment:
This hides the exception stacktrace. Maybe not the biggest issue here since it's unlikely that line 169 will recurse very deeply, but we do this kind of "catch, log and continue" thing in quite a number of places and therefore it might be worthwhile to factor it out into a dedicated function? Here's my proposal:
```julia
function log_errors(f)
    try
        f()
    catch
        io = IOBuffer()
        # Copied from https://github.com/JuliaLang/julia/blob/9aba4a8db97771e49d7b99bc2140780a1dd49a9c/base/client.jl#L112
        Base.show_exception_stack(IOContext(io, :limit => true), Base.catch_stack())
        @error String(take!(io))
    end
end
```

Example usage:
```julia
julia> log_errors() do
           error("boo!")
       end
┌ Error: boo!
│ Stacktrace:
│   [1] error(s::String)
│     @ Base ./error.jl:33
│   [2] (::var"#1#2")()
│     @ Main ./REPL[2]:2
│   [3] log_errors(f::var"#1#2")
│     @ Main ./REPL[1]:3
│   [4] top-level scope
│     @ REPL[2]:1
│   [5] eval
│     @ ./boot.jl:360 [inlined]
│   [6] eval_user_input(ast::Any, backend::REPL.REPLBackend)
│     @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:139
│   [7] repl_backend_loop(backend::REPL.REPLBackend)
│     @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:200
│   [8] start_repl_backend(backend::REPL.REPLBackend, consumer::Any)
│     @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:185
│   [9] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool)
│     @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:317
│  [10] run_repl(repl::REPL.AbstractREPL, consumer::Any)
│     @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:305
│  [11] (::Base.var"#881#883"{Bool, Bool, Bool})(REPL::Module)
│     @ Base ./client.jl:387
│  [12] #invokelatest#2
│     @ ./essentials.jl:708 [inlined]
│  [13] invokelatest
│     @ ./essentials.jl:706 [inlined]
│  [14] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
│     @ Base ./client.jl:372
│  [15] exec_options(opts::Base.JLOptions)
│     @ Base ./client.jl:302
│  [16] _start()
│     @ Base ./client.jl:485
└ @ Main REPL[1]:8
```

I've verified that this works on both Julia v1.6 and v1.10.

Reply:
We have a `reporterror()` which does not serve the same purpose, but is closely related. Perhaps we can combine that in here too, so we have something like:
```julia
reporterrors(...) do
  ...
end
```
where some of the parameters to `reporterror()` can optionally be passed in too, with a common syntax. If we agree on this, I can do a mass fix for this.

Reply:
I forgot about `reporterror()`, and I agree that this should be taken into consideration here. I guess the `reporterrors()` implementation could just be this: 
```julia
function reporterrors(f, src)
    try
        f()
    catch
        reporterror(src, Base.catch_stack()) # ToDo: Check that catch_stack() is the correct function to call here
    end
end
```
However, I'm not very fond of `reporterror()` printing the stacktrace in a custom way. The Julia community has worked and is working hard to make stacktraces as easy to read as possible, and the current form of `reporterror()` basically sidesteps all of these efforts. Maybe `reporterror()` could be something along these lines?
```julia
function reporterror(src, err)
    io = IOBuffer()
    println(io, "Error in $src")
    Base.show_exception_stack(IOContext(io, :limit => true), err)
    @error String(take!(io))
end
```

Reply:
`reporterror()` was implemented because Julia's exception stack traces has become unusable with the nesting level in fjage. It has certainly improved now, and we can switch to it. It just means that on older Julia, the stack traces will be bad, but if we are mostly on newer Julia versions, we should be good.

Reply:
How about offering 2 functions, to be used depending on need...?
```julia
reporterrors(src) do
  # do naughty things here that might throw an error
end

try
  # do naughty things here that might throw an error
catch ex
  reporterror(src, ex)
  # do something more to handle the error
end
```
We can go with your implementation of `reporterror()` that uses `show_exception_stack()`.

Reply:
Good question – I don't recall details. But I am happy to go with your suggested reporting based on the improved Julia API, so this code will go away anyway.

Reply:
I will refactor shortly based on `logerror()` as agreed.

Reply:
Done. @ettersi, do take a look if it looks OK.

Reply:
This version of `logerror(::Exception, ...)` prints only the most recent error: 
```julia
julia> Fjage.logerror() do
           try
               error("original error")
           catch
               error("error while handling original error")
           end
       end
┌ Error: error while handling original error
│ Stacktrace:
│   [1] error(s::String)
│     @ Base ./error.jl:35
│   [2] (::var"#3#4")()
│     @ Main ./REPL[4]:5
│   [3] logerror(f::var"#3#4", src::Nothing)
│     @ Fjage ~/Documents/Fjage.jl/src/container.jl:712
│   [4] logerror(f::Function)
│     @ Fjage ~/Documents/Fjage.jl/src/container.jl:711
│   [5] top-level scope
│     @ REPL[4]:1
│   [6] eval
│     @ Core ./boot.jl:385 [inlined]
│   [...]
└ @ Fjage ~/Documents/Fjage.jl/src/container.jl:728
```
Assuming we don't care about manipulating the exception object, may I suggest this implementation?
```julia
julia> function logerror(f::Function, src=nothing)
         try
           f()
         catch
           logerror(src)
         end
       end;

julia> function logerror(src=nothing)
         io = IOBuffer()
         src === nothing || print(io, "[$src] ")
         Base.show_exception_stack(IOContext(io, :limit => true), Base.catch_stack())
         @error String(take!(io))
       end;

julia> logerror() do
           try
               error("original error")
           catch
               error("error while handling original error")
           end
       end
┌ Error: error while handling original error
│ Stacktrace:
│   [1] error(s::String)
│     @ Base ./error.jl:35
│   [2] (::var"#5#6")()
│     @ Main ./REPL[7]:5
│   [3] logerror(f::var"#5#6", src::Nothing)
│     @ Main ./REPL[5]:3
│   [4] logerror(f::Function)
│     @ Main ./REPL[5]:2
│   [5] top-level scope
│     @ REPL[7]:1
│   [6] eval
│     @ Core ./boot.jl:385 [inlined]
│   [...]
│ 
│ caused by: original error
│ Stacktrace:
│   [1] error(s::String)
│     @ Base ./error.jl:35
│   [2] (::var"#5#6")()
│     @ Main ./REPL[7]:3
│   [3] logerror(f::var"#5#6", src::Nothing)
│     @ Main ./REPL[5]:3
│   [4] logerror(f::Function)
│     @ Main ./REPL[5]:2
│   [5] top-level scope
│     @ REPL[7]:1
│   [6] eval
│     @ Core ./boot.jl:385 [inlined]
│   [...]
└ @ Main REPL[6]:5
```

Reply:
@mchitre Was this missed or ignored? Latter is fine, just making sure it's not the former.

Reply:
Missed somehow. Will get to it in a bit...

Reply:
Fixed directly on master, since this was merged. Do take a look @ettersi.

Reply:
The exception argument needed to be removed from the calls to `logerror()`. I also updated directly on master: https://github.com/org-arl/Fjage.jl/commit/e0d494ccd7ede8ea5632efd461d4fb16e0fe317c
Looks good otherwise.